internal/validation/k8s/objectmeta.go

```diff
@@ -20,28 +20,27 @@ limitations under the License.
 package k8s
 
 import (
+	"errors"
 	"fmt"
 	"strings"
-
-	"tags.cncf.io/container-device-interface/internal/multierror"
 )
 
 // TotalAnnotationSizeLimitB defines the maximum size of all annotations in characters.
 const TotalAnnotationSizeLimitB int = 256 * (1 << 10) // 256 kB
 
 // ValidateAnnotations validates that a set of annotations are correctly defined.
 func ValidateAnnotations(annotations map[string]string, path string) error {
-	errors := multierror.New()
+	errs := []error{}
 	for k := range annotations {
 		// The rule is QualifiedName except that case doesn't matter, so convert to lowercase before checking.
 		for _, msg := range IsQualifiedName(strings.ToLower(k)) {
-			errors = multierror.Append(errors, fmt.Errorf("%v.%v is invalid: %v", path, k, msg))
+			errs = append(errs, fmt.Errorf("%v.%v is invalid: %v", path, k, msg))
 		}
 	}
 	if err := ValidateAnnotationsSize(annotations); err != nil {
-		errors = multierror.Append(errors, fmt.Errorf("%v is too long: %v", path, err))
+		errs = append(errs, fmt.Errorf("%v is too long: %v", path, err))
 	}
-	return errors
+	return errors.Join(errs...)
```

Comment:
```suggestion
	return errs
```

Reply:
That is not the same.

While I realize that from a pure compilability/errors API point of view it could be implemented like that, I deliberate chose not to do it.

We've converted a number of packages from multierror (both the original hashicorp one and one-off imitation like this) and in every conversion so far we've tried to consistently follow this same pattern: keep errors that originate at the same 'level' also at the same level in the joint error.

If you look at the errors.Join, you'll see that your suggested changes end up with a very different internal representation. Instead of a flat joinError with a single slice containing all the errors at the same logical level, you end up with deeply nested (sort of unbalanced tree) of joinErrors, with each one containing a single error as errs[0] and a similarly recursive jointError as errs[1].